### ISSUE TITLE
bootstrap update notifications: only show what's new

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#122344]
Bootstrap now remembers which "update warnings" it already showed me, and so when I run it the second time it doesn't keep repeating the same messages. That's great. :)

However, when something does change, it prints more than I would expect:
```
There have been changes to x.py since you last updated:
  [WARNING] The "codegen"/"llvm" profile has been removed and replaced with "compiler", use it instead for the same behavior.
    - PR Link https://github.com/rust-lang/rust/pull/121278
  [INFO] `x install` now skips providing tarball sources (under 'build/dist' path) to speed up the installation process.
    - PR Link https://github.com/rust-lang/rust/pull/118724
  [INFO] A new `boostrap-cache-path` option has been introduced which can be utilized to modify the cache path for bootstrap.
    - PR Link https://github.com/rust-lang/rust/pull/121976
  [INFO] a new `target.*.runner` option is available to specify a wrapper executable required to run tests for a target
    - PR Link https://github.com/rust-lang/rust/pull/122108
  [INFO] New option `rust.llvm-bitcode-linker` that will build the llvm-bitcode-linker.
    - PR Link https://github.com/rust-lang/rust/pull/117458
NOTE: to silence this warning, update `config.toml` to use `change-id = 117458` instead
```
At least the first 2 I have already seen, but somehow it keeps repeating them every time a new item is added to the list.

Is it possible that when computing which thing to show, it uses the value in the `config.toml` rather than the value in the "last things I showed" file? If so, would it be possible to only show what's new since the last time something was shown? That way I don't have to re-read the same update notifications over and over again, and the list does not keep growing indefinitely. :)

Cc @onur-ozkan
